### ISSUE TITLE
Use with other bootloaders?

### ISSUE THREAD

**Issue by ghost** [scwimbush/Printrboard-HID-Arduino-IDE-Support#2]
Would it be possible to use this to compile a .hex and then upload it to a Printrboard that's running CDC bootloader with something like AVRdude?  Or if you have a bit, maybe point me in a direction for getting it to work with CDC bootloader?

**Comment by scwimbush** [comment]
I believe this will do what you suggest (the instructions on the RepRapWiki tell you where to find the .hex file), but why not just use the package that's set up for the CDC bootloader already? Sorry I don't have time to create (or test) a CDC version. It doesn't interest me since it will always have the 64k limit and my firmware is larger. If you want to try it yourself, you should be able to modify the files in the existing distribution to match the pattern of these ones.

**Comment by ghost** [comment]
I ended up using an Arduino I had around as an ISP to flash the HID firmware.  We're still limited to 128k for the firmware size right?  Have any suggestions what to trim out from the current marlin release, I believe it compiled to 207k.

**Comment by scwimbush** [comment]
I'm glad to hear you had success with the HID bootloader. Once you've successfully flashed the bootloader, it really is the simplest method.

The firmware size is still limited to 128k by the amount of flash memory available on the chip, so this is a hard limit. I'm surprised to hear that your firmware compiles to 207k -- that seems very large. I don't use Marlin, but most firmwares I know of were originally able to fit in 64k, and mine has just crept above that size with all features enabled -- about 70k. The things I know of that pushed the size up were the SD card support and the graphical (LCD) display support, so you could try disabling those if you don't need them.

Are you sure it is the compiled size that is 207k? The .hex file will be larger than the binary size, which is the number reported by the Arduino IDE when the firmware is compiled. Perhaps it is your .hex file that is 207k, in which case the binary size might be about half that.

**Comment by ghost** [comment]
Yeah, I was looking at the file size, it was 207k.  Figured the actual firmware would be smaller.  So I tried to flash it anyway and found out that your package doesn't work on Mac.  Because the hid_bootloader_cli included is a Windows executable.  Tried to find it already compiled for Mac but got wierd errors when I tried with it.  Booted into Windows and your package works fine.  My .hex file flashed just fine.  Looks like I'll have to compile hid_bootloader_cli myself for Mac.

**Comment by scwimbush** [comment]
Thanks for sharing this detailed info. I must confess I don't know how Arduino works cross-platform, but if you figure out a way to get it working on Mac, you can certainly fork this for Mac, or it can be incorporated directly here. Glad to get the first third-party report of success on Windows.

**Comment by ghost** [comment]
Well, I can't connect to the board.  Arduino seems to upload Marlin to it just fine, just won't connect to Repetier

**Comment by ghost** [comment]
Blah, looks like it was an issue with current Marlin, actual stable release worked.
